### ISSUE TITLE
Implicit fall-through warning while using gcc 7.2

### ISSUE THREAD

**Issue by sickpig** [gandrewstone/bitcoin#961]
While compiling the source code using gcc 7.2 I noticed that there's a few places in the code that produce warnings like this one:

```
gcc -I. -g -O2 -Wall -Wextra -Wno-unused-function -c src/gen_context.c -o gen_context.o
hash.cpp: In function 'unsigned int MurmurHash3(unsigned int, const std::vector<unsigned char>&)':
hash.cpp:50:16: warning: this statement may fall through [-Wimplicit-fallthrough=]
             k1 ^= tail[2] << 16;
             ~~~^~~~~~~~~~~~~~~~
hash.cpp:51:9: note: here
         case 2:
         ^~~~
hash.cpp:52:16: warning: this statement may fall through [-Wimplicit-fallthrough=]
             k1 ^= tail[1] << 8;
             ~~~^~~~~~~~~~~~~~~
hash.cpp:53:9: note: here
```

basically what's happening is described here:
https://developers.redhat.com/blog/2017/03/10/wimplicit-fallthrough-in-gcc-7/ (edited)

To solve the issue we need to explicitly mark the branches where it is intended to fall through the next one.

**Comment by sickpig** [comment]
this issue should be solved by #962

**Comment by sickpig** [comment]
closing since #962 has been merged
